sql/rewards/rewards.go

```diff
@@ -7,79 +7,45 @@ import (
 	"github.com/spacemeshos/go-spacemesh/sql"
 )
 
-const (
-	coinbaseReward = iota + 1
-	smesherReward
-)
-
 // Add reward to the database.
-func Add(db sql.Executor, lid types.LayerID, reward *types.AnyReward) error {
-	if _, err := db.Exec(`insert into rewards 
-			(smesher, coinbase, layer, total_reward, layer_reward) 
-			values (?1, ?2, ?3, ?4, ?5) 
-		on conflict(smesher,layer) 
-			do update set 
-				total_reward=add_uint64(total_reward, ?4),
-		 		layer_reward=add_uint64(layer_reward, ?5);`,
+func Add(db sql.Executor, reward *types.Reward) error {
+	if _, err := db.Exec(`insert into rewards (coinbase, layer, total_reward, lyr_reward) values (?1, ?2, ?3, ?4);`,
 		func(stmt *sql.Statement) {
-			stmt.BindBytes(1, reward.SmesherID.ToBytes())
-			stmt.BindBytes(2, reward.Address[:])
-			stmt.BindInt64(3, int64(lid.Uint32()))
-			stmt.BindInt64(4, int64(reward.Amount))
-			stmt.BindInt64(5, int64(reward.LayerReward))
+			stmt.BindBytes(1, reward.Coinbase[:])
+			stmt.BindInt64(2, int64(reward.Layer.Uint32()))
+			stmt.BindInt64(3, int64(reward.TotalReward))
+			stmt.BindInt64(4, int64(reward.LayerReward))
 		}, nil); err != nil {
-		return fmt.Errorf("insert %s %+x: %w", lid, reward, err)
+		return fmt.Errorf("insert %+x: %w", reward, err)
 	}
 	return nil
 }
 
-// order of fields - smesher, coinbase, layer, total_reward, layer_reward .
-func decodeReward(stmt *sql.Statement) (types.Reward, error) {
-	reward := types.Reward{
-		Layer:               types.NewLayerID(uint32(stmt.ColumnInt64(2))),
-		TotalReward:         uint64(stmt.ColumnInt64(3)),
-		LayerRewardEstimate: uint64(stmt.ColumnInt64(4)),
-	}
-	stmt.ColumnBytes(0, reward.SmesherID[:])
-	stmt.ColumnBytes(1, reward.Coinbase[:])
-	return reward, nil
-}
-
-// FilterByCoinbase filters rewards from all layers by coinbase address.
-func FilterByCoinbase(db sql.Executor, address types.Address) (rst []types.Reward, err error) {
-	if _, err := db.Exec("select smesher, coinbase, layer, total_reward, layer_reward from rewards where coinbase = ?1;",
+// RevertToLayer reverts the rewards to the specified layer.
+func RevertToLayer(db sql.Executor, revertTo types.LayerID) error {
```

Comment:
can we write more compact apis -> Revert(db, to Layer)? why it needs to have same stuff repeated multiple times